enhancements/olm/safe-cluster-upgrades.md

```diff
@@ -0,0 +1,267 @@
+authors:
+  - "@awgreene"
+reviewers:
+  - TBD
+  - "@kevinrizza"
+approvers:
+  - "@ecordell"
+  - "@spadgett"
+creation-date: 2021-01-11
+last-updated: 2021-01-19
+status: provisional
+see-also:
+  -  N/A
+replaces:
+  - N/A
+superseded-by:
+  - N/A
+---
+
+# Safe OpenShift Upgrades with Operators
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Definitions
+
+**OLM Managed Operators**: Operators that are installed, managed, and upgraded by the [Operator Lifecycle Manager (OLM)](https://github.com/operator-framework/operator-lifecycle-manager) project.
+
+**Minor OpenShift Upgrade**: When the OpenShift Cluster is upgraded to the next minor version, for example the upgrade from version 4.6 to 4.7. See [Semantic Versioning](https://semver.org/) for more details.
+
+## Summary
+
+OpenShift Cluster Admins introduce new services to their clusters in the form of OLM Managed Operators. When performing a cluster upgrade, there is no guarantee that these operators will continue to run on the upgraded cluster.
+
+OLM must make an effort to prevent cluster upgrades when it is possible to determine that one or more installed operators will not run on the next version of OpenShift.
+
+## Motivation
+
+The primary purpose of this enhancement is to define how OpenShift can protect existing services introduced by OLM Managed Operators when performing a Minor OpenShift upgrade.
+
+### Goals
+
+- Prevent Minor OpenShift Upgrades when OLM detects that an installed operator will not run on the next Minor OpenShift version.
+
+### Non-Goals
+
+- Ensure that services will not be disrupted during a cluster upgrade.
+- Guarantee that all installed operator will continue to run after a cluster upgrade.
+- Prevent Patch OpenShift Upgrades.
+
+## Proposal
+
+The [ClusterVersionOperator](https://github.com/openshift/cluster-version-operator) provides OLM with the means to prevent Minor Semantic Version cluster upgrades by setting the  `operator-lifecycle-manager` [ClusterOperator's](https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/clusteroperator.md) [upgradeable condition](https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/clusteroperator.md#what-should-an-operator-report-with-clusteroperator-custom-resource) to `False`.
+
+OLM currently reports that it is upgradeable as soon as the OLM Deployment is successfully rolled out. This existing logic should be updated to monitor the collection of OLM Managed Operators on cluster and update the `upgradeable condition` to reflect if any of these operators will not run on the next Minor OpenShift Version.
```

Comment:
```suggestion
OLM currently reports that it is upgradeable as soon as the OLM Deployment is successfully rolled out. This existing logic should be updated to monitor the collection of OLM Managed Operators on cluster and update the `upgradeable condition` to reflect if any of these operators indicate that they will not run on the next Minor OpenShift Version.
```

(TBD: what we want to do if the operator gives no indication either way)

Reply:
I accepted the suggested changes, please mark the conversation as resolved if applicable.